File: src/main/java/no/nav/helse/flex/operations/journalforing/JournalforingOperations.java
package no.nav.helse.flex.operations.journalforing;

import no.nav.helse.flex.infrastructure.kafka.EnrichedKafkaEvent;
import no.nav.helse.flex.operations.journalforing.dokarkiv.JournalpostAPIClient;
import no.nav.helse.flex.infrastructure.exceptions.ExternalServiceException;
import no.nav.helse.flex.infrastructure.exceptions.TemporarilyUnavailableException;
import no.nav.helse.flex.operations.eventenricher.journalpost.Journalpost;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class JournalforingOperations {
    private static final Logger log = LoggerFactory.getLogger(JournalforingOperations.class);
    private final JournalpostAPIClient journalpostAPIClient;

    public JournalforingOperations(){
        this.journalpostAPIClient = new JournalpostAPIClient();
    }

    public void doAutomaticStuff(final EnrichedKafkaEvent enrichedKafkaEvent) throws TemporarilyUnavailableException, ExternalServiceException {
        enrichedKafkaEvent.getJournalpost().updateWithGenerellSak();
        populateAvsendeMottaker(enrichedKafkaEvent);

        // TODO: Slå på når vi skal behandle
        journalpostAPIClient.updateJournalpost(enrichedKafkaEvent.getJournalpost());

        log.info("Oppdatert {}", enrichedKafkaEvent.getJournalpost().toString());
        journalpostAPIClient.finalizeJournalpost(enrichedKafkaEvent.getJournalpostId());
        log.info("Ferdigstilt og fullført behandling av journalpost {}.", enrichedKafkaEvent.getJournalpostId());
    }

    private void populateAvsendeMottaker(final EnrichedKafkaEvent enrichedKafkaEvent) {
        final Journalpost journalpost = enrichedKafkaEvent.getJournalpost();
        if (journalpost.getAvsenderMottaker() == null || journalpost.getAvsenderMottaker().getId() == null) {
            journalpost.settAvsenderMottaker(enrichedKafkaEvent.getFnr(), "FNR");
            log.info("Setter bruker som avsender på journalpost: {}", journalpost.getJournalpostId());
        }
    }
}

File: src/main/kotlin/no/nav/helse/flex/operations/journalforing/JournalforingOperations.kt
package no.nav.helse.flex.operations.journalforing

import no.nav.helse.flex.infrastructure.kafka.EnrichedKafkaEvent
import no.nav.helse.flex.operations.eventenricher.journalpost.Journalpost
import no.nav.helse.flex.operations.journalforing.dokarkiv.JournalpostAPIClient
import org.slf4j.LoggerFactory

class JournalforingOperations(
    private val journalpostAPIClient: JournalpostAPIClient = JournalpostAPIClient()
) {
    private val log = LoggerFactory.getLogger(JournalforingOperations::class.java)

    fun doAutomaticStuff(enrichedKafkaEvent: EnrichedKafkaEvent) {
        enrichedKafkaEvent.journalpost!!.updateWithGenerellSak()
        populateAvsendeMottaker(enrichedKafkaEvent.journalpost!!, enrichedKafkaEvent.fnr!!)
        updateJournalpost(enrichedKafkaEvent.journalpost!!)
        finalizeJournalpost(enrichedKafkaEvent.journalpostId)
    }

    private fun populateAvsendeMottaker(journalpost: Journalpost, fnr: String) {
        if (journalpost.avsenderMottaker?.id == null) {
            journalpost.settAvsenderMottaker(fnr, "FNR")
            log.info("Setter bruker som avsender på journalpost: ${journalpost.journalpostId}")
        }
    }

    private fun updateJournalpost(journalpost: Journalpost) {
        journalpostAPIClient.updateJournalpost(journalpost)
        log.info("Oppdatert $journalpost")
    }

    private fun finalizeJournalpost(journalpostId: String) {
        journalpostAPIClient.finalizeJournalpost(journalpostId)
        log.info("Ferdigstilt og fullført behandling av journalpost $journalpostId.")
    }
}
